src/client/async/async.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Use of this source code is governed by The MIT License.
+// See the LICENSE file for details.
+
+/**
+ * @fileoverview Fast asynchronous function execution.
+ *
+ * This package provides functions to defer execution on the main thread
+ * without using setTimeout, though setTimeout is used as a fallback in browsers
+ * that do not support other methods.  Using these methods is advantageous when
+ * one wants to schedule a callback faster than the setTimeout clamped minimum
+ * allows (e.g. when doing `setTimeout(fn, 0)`)  The clamped minimum for
+ * setTimeout is often 10ms, though when WebKit browsers are in a background
+ * tab, setTimeout calls deprioritized to execute with a 1s delay.  In these
+ * cases, this package provides an alternative.
+ *
+ * @author nicksay@google.com (Alex Nicksay)
+ */
+
+goog.provide('spf.async');
+
+goog.require('spf');
+goog.require('spf.state');
+goog.require('spf.tracing');
+
+
+
+/**
+ * Initializes fast asynchronous function execution.
+ */
+spf.async.init = function() {
+  if (!spf.state.get(spf.state.Key.ASYNC_INIT)) {
+    if (spf.async.CAN_USE_POSTMESSAGE_) {
+      spf.async.addListener_(spf.async.handleMessage_);
+      spf.state.set(spf.state.Key.ASYNC_LISTENER, spf.async.handleMessage_);
+    }
+    spf.state.set(spf.state.Key.ASYNC_INIT, true);
+  }
+};
+
+
+/**
+ * Disposes fast asynchronous function execution.
+ */
+spf.async.dispose = function() {
+  if (spf.state.get(spf.state.Key.ASYNC_INIT)) {
+    spf.async.removeListener_(/** @type {function(Event)} */ (
+          spf.state.get(spf.state.Key.ASYNC_LISTENER)));
+    spf.state.set(spf.state.Key.ASYNC_INIT, false);
+    spf.state.set(spf.state.Key.ASYNC_LISTENER, null);
```

Comment:
Not blocking but I think this would be a good use case for a spf.state.clear method.